projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-admin-menu.php

```diff
@@ -368,28 +368,16 @@ private function parse_count_data( $item ) {
 		$title = $item['title'];
 
 		if ( false !== strpos( $title, 'count-' ) ) {
-			preg_match( '/class="(.+\s)?count-(\d*)/', $title, $matches );
+			preg_match( '/<span class=".+\s?count-(\d*).+\s?<\/span><\/span>/', $title, $matches );
 
-			$count = absint( $matches[2] );
+			$count = absint( $matches[1] );
 			if ( $count > 0 ) {
+				// Keep the counter in the item array.
 				$item['count'] = $count;
 			}
-		}
-
-		return $item;
-	}
-
-	/**
-	 * Removes unexpected markup from the title.
-	 *
-	 * @param array $item containing title to parse.
-	 * @return array
-	 */
-	private function sanitize_title( $item ) {
-		$title = $item['title'];
 
-		if ( wp_strip_all_tags( $title ) !== trim( $title ) ) {
-			$item['title'] = trim( substr( $title, 0, strpos( $title, '<' ) ) );
+			// Finally remove the markup.
+			$item['title'] = trim( str_replace( $matches[0], '', $title ) );
```

Comment:
I found a bit surprising that this function modifies the menu title, since it's not immediately obvious from the `parse_count_data` name (and it's also not documented in the docstring of the function).

What if we rename the function to something more abstract like `parse_menu_item`, which also sanitizes the menu title?

Reply:
That's a good point, I took the liberty in consolidating `parse_count_data` and `parse_markup_data` in `parse_menu_item` https://github.com/Automattic/jetpack/pull/18924/commits/9c23c5ffd04b54af253a4b246b6bccaa53b97241